### PR TITLE
add ability to specify heap size hint as a percent

### PR DESCRIPTION
**Pull request by JeffBezanson**
also fix overflowed subtraction with size hints < 250mb --- we should pick some minimum soft heap limit; I chose 1mb here but it could be anything.

### CODE REVIEW

File: doc/man/julia.1 [hunk 1]
```diff
@@ -227,8 +227,8 @@ fallbacks to the latest compatible BugReporting.jl if not. For more information,
 
 .TP
 --heap-size-hint=<size>
-Forces garbage collection if memory usage is higher than that value. The memory hint might be
-specified in megabytes (500M) or gigabytes (1.5G)
+Forces garbage collection if memory usage is higher than that value. The value can be
+specified in units of K, M, G, T, or %.
```
Comment (vtjnash) [inline doc/man/julia.1:231]:
```suggestion
specified in units of K, M, G, T, or % of physical memory.
```

**Review by vtjnash — COMMENTED** [review]
LGTM

**Review by gbaraldi — APPROVED** [review]

**Comment by vtjnash** [comment]
No, it failed on CI:
```

2024-02-01 04:27:38 UTC	Error in testset cmdlineargs:
2024-02-01 04:27:38 UTC	Test Failed at C:\buildkite-agent\builds\win2k22-amdci6-2\julialang\julia-master\julia-ec4182e384\share\julia\test\cmdlineargs.jl:1059
2024-02-01 04:27:38 UTC	  Expression: abs(maxmem - (hint - 250 * 1024 * 1024)) / maxmem < 0.01
2024-02-01 04:27:38 UTC	   Evaluated: 4.170485178629558 < 0.01
```

**Comment by JeffBezanson** [comment]
Seems I did not account for 32-bit machines with 32gb of memory :thinking: :confused:

**Comment by vtjnash** [comment]
Nobody ever does, until they realize it is probably the most common 32 bit machines now, haha

**Pull request merged**